### ISSUE TITLE
Bug in: ClassAttributeWithStaticValue

### ISSUE THREAD

**Issue by dennisvandehoef** [brigade/haml-lint#100]
this code:

```haml
- unless @show_discount == 'dont_show'
      .discount{ class: @show_discount }
```

throws the error: `ClassAttributeWithStaticValue: Avoid defining `class` in attributes hash for static class names`

When we change it to this (add to_s) haml-lint sees that it is a variable containing a string and accepts it

```haml
- unless @show_discount == 'dont_show'
      .discount{ class: @show_discount.to_s }
```

**Comment by sonalkr132** [comment]
[This line](https://github.com/brigade/haml-lint/blob/d7e3c43ef02090bd4c88f6fa65da189ce7eed1cc/lib/haml_lint/linter/class_attribute_with_static_value.rb#L48) is causing the issue.
Instance variable gets parsed as `(ivar :@show_discount)`. So if you do `value.children`  it would return `[:@show_discount]`.

> When we change it to this (add to_s) haml-lint sees that it is a variable containing a string and accepts it

`@show_discount.to_s` gets parsed as function call over ivar. Thats why it is passing. 

Local variables are also passing the test cause they are getting parsed as method calls (oh the ambiguity!! :sweat: ). ie `(send nil :show_discount)`. So here `value.children.first.class` would return nilClass. ( `STATIC_CLASSES.include?` => false). It might be worth finding out how rubocop is handling this.

I think a quick fix can be that `value.children.first.class` is replaced with, `value.type` instead.
String and Symbol has `value.type` of `str` and `sym` respectively. Is it acceptable?

**Comment by sds** [comment]
Thanks for the report @dennisvandehoef, and thank you for investigating @sonalkr132. Your suggested solution sounds great, and I've gone ahead and implemented it in 5b0af87d17b23fc06620a21c17f27e64a1682489.

Thanks for your help!
